storage_controller/src/http.rs

```diff
@@ -254,6 +296,7 @@ async fn handle_tenant_config_get(
 ) -> Result<Response<Body>, ApiError> {
     let tenant_id: TenantId = parse_request_param(&req, "tenant_id")?;
     check_permissions(&req, Scope::PageServerApi)?;
+    maybe_rate_limit(tenant_id).await;
```

Comment:
It looks to me like the request futures are dropped on shutdown, so we don't need any explicit cancellation handling here.